src/app/components/table/table.ts

```diff
@@ -5856,6 +5903,8 @@ export class ColumnFilter implements AfterContentInit {
     }
 
     applyFilter() {
+        // deep copy of current applied filter
+        this.appliedFilter = JSON.parse(JSON.stringify(this.dt.filters[<string>this.field]));
```

Comment:
Changes to the applyFilter method will not fix the case where the filter is saved not using the "apply" filter button.  

I believe this change (note:  I have not verified this using your branch) will not fix Issue #15778.  Please see Issue #15778.  I added a video that demonstrates the issue.  In Issue #15778 it describes the steps to reproduce the bug in the genral bug description.

If the filter is a multi-select filter,  the apply button will not be called and these changes will have no effect.  At the very end of the video that I include for issue #15804,  the advance filters are broken using a multi-select filter and the apply button is not pressed.

I have not verified the issue with all advance filters in the demo and  "beyond"